### PR TITLE
feat: add sponsor modal

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Add modal to show when user add first bookmark

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://mi-663-folder-sponsor-modal.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          <Button
+            className="self-stretch"
+            icon={<ArrowIcon className="rotate-90" />}
+            iconPosition={ButtonIconPosition.Right}
+            variant={ButtonVariant.Primary}
+          >
+            Create 1 for free
+          </Button>
+          <Link passHref href={plusUrl}>
+            <Typography
```
Comment (AmarTrebinjac) [inline packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:96]: Should be  `type={TypographyType.Callout}`

**Review by AmarTrebinjac — CHANGES_REQUESTED** [review]
Looks good! 🤜🏻🤛🏻

Couple of minor comments

File: packages/extension/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "extension",
-  "version": "3.35.7",
+  "version": "3.35.8",
```
Comment (rebelchris) [inline packages/extension/package.json:3]: How did this change here? :jenkins-triggered:

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import { ActionType } from '../../../graphql/actions';
+import { Button } from '../../buttons/Button';
+import { ArrowIcon } from '../../icons';
+import { plusUrl } from '../../../lib/constants';
+import { LogEvent, TargetId } from '../../../lib/log';
+import Link from '../../utilities/Link';
+
+type BookmarkFolderSponsorModalProps = Omit<ModalProps, 'children'>;
+
+export const BookmarkFolderSponsorModal = (
```
Comment (rebelchris) [inline packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:28]: Why Sponsor? 
BookmarksFolderEntryModal is maybe more accurate?

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  const isTablet = useViewSize(ViewSize.Tablet);
+
+  useEffect(() => {
+    if (
+      isActionsFetched &&
+      !checkHasCompleted(ActionType.CreateBookmarkFolder)
+    ) {
+      completeAction(ActionType.CreateBookmarkFolder);
+    }
+  }, [checkHasCompleted, completeAction, isActionsFetched]);
```
Comment (rebelchris) [inline packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:43]: I don't think you need a useEffect here?
Either just fire it all the time or simple inline check it's not set yet.

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+  return (
+    <Modal
+      kind={Modal.Kind.FlexibleCenter}
+      size={Modal.Size.Small}
+      isDrawerOnMobile
+      {...props}
+    >
+      <ModalClose
+        className="right-4 top-4"
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:53]: It's fine mostly I think, design close buttons have been all over the place lateltly

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        onClick={onRequestClose}
+        variant={ButtonVariant.Primary}
+      />
+      <Modal.Body className="!tablet:p-4 flex flex-col justify-center gap-4 tablet:items-center tablet:text-center">
```
Comment (rebelchris) [inline packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:57]: Do we need the special tablet changes? what's it for?
Normally our modals are pretty standarized.

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+          <Image
+            className="rounded-16"
+            src={bookmarkFolderSoonImage}
+            alt="Bookmark Folder coming soon"
+          />
+        </div>
+        <Typography
+          bold
+          color={TypographyColor.Primary}
+          type={isTablet ? TypographyType.Title1 : TypographyType.LargeTitle}
```
Comment (rebelchris) [inline packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:68]: This is also new, haven't seen it on a modal yet I don't think?

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+          </Button>
+          <Link passHref href={plusUrl}>
+            <Typography
+              bold
+              className="underline hover:no-underline"
+              color={TypographyColor.Plus}
+              onClick={() => {
+                logSubscriptionEvent({
+                  event_name: LogEvent.UpgradeSubscription,
+                  target_id: TargetId.BlockedWords,
```
Comment (rebelchris) [inline packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:103]: Wrong targetId here

**Review by rebelchris — CHANGES_REQUESTED** [review]
Couple minor things

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx (hunk 6 above)
Comment (ilasw) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:57]: On mobile we have items justified to the left and from tablet they are all centered 👀

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx (hunk 7 above)
Comment (ilasw) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:68]: Should I ask to design to change it? This is from figma

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        <div className="flex flex-col-reverse items-center gap-3 self-stretch tablet:flex-col tablet:gap-4">
+          <Button
+            className="self-stretch"
+            icon={<ArrowIcon className="rotate-90" />}
+            iconPosition={ButtonIconPosition.Right}
+            variant={ButtonVariant.Primary}
+          >
+            Create 1 for free
+          </Button>
+          <Link passHref href={plusUrl}>
```
[the remark this answers is not in the document]
Comment (ilasw) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:95]: But I'm using the button where I've the button and the link where the link is. What are you pointing at? 👀

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx (hunk 7 above)
Comment (rebelchris) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx:68]: Hmm yeah just verify if it's intended and we normally don't do this?

File: packages/extension/package.json (hunk 2 above)
Comment (ilasw) [reply packages/extension/package.json]: Target branch didn't have it, just update and now is fine ✅

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx (hunk 4 above)
Comment (ilasw) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx]: Just didn't want to check for every render, I think is safer limiting it. Wdyt?

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx (hunk 7 above)
Comment (ilasw) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx]: Design want to keep those changes ✔️

File: packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx (hunk 6 above)
Comment (ilasw) [reply packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx]: Design removed those ✔️

**Review by rebelchris — APPROVED** [review]
Non blocking, but think we can get rid of useEffect

**Review by AmarTrebinjac — APPROVED** [review]

**Pull request merged**